src/Pathy/Printer.purs

```diff
@@ -0,0 +1,182 @@
+module Pathy.Printer
+  ( Printer
+  , posixPrinter
+  , windowsPrinter
+  , printPath
+  , unsafePrintPath
+  , debugPrintPath
+  , Escaper(..)
+  , slashEscaper
+  , dotEscaper
+  , posixEscaper
+  , windowsEscaper
+  ) where
+
+import Prelude
+
+import Data.Foldable (fold)
+import Data.Maybe (Maybe, maybe)
+import Data.Monoid (class Monoid)
+import Data.Newtype (class Newtype, un, unwrap)
+import Data.String as Str
+import Data.String.NonEmpty (NonEmptyString)
+import Data.String.NonEmpty as NES
+import Partial.Unsafe (unsafePartial)
+import Pathy.Path (Path, canonicalize, foldPath, (</>))
+import Pathy.Phantom (class IsDirOrFile, class IsRelOrAbs, foldDirOrFile, foldRelOrAbs, onRelOrAbs)
+import Pathy.Sandboxed (SandboxedPath, sandboxRoot, unsandbox)
+
+-- | A `Printer` defines options for printing paths.
+-- |
+-- | - `root` is a function used to construct the initial segment of paths.
+-- | - `current` is a representation of the current directory.
+-- | - `up` is a representation of going up to the parent directory.
+-- | - `sep` is the string to separate path segments by.
+-- | - `escaper` specified how to deal with printing reserved names and
+-- |    characters.
+type Printer =
+  { root :: Maybe NonEmptyString -> String
```

Comment:
This root thing is not used during print so why do we need it?

Reply:
Oh, that's a mistake... hmm. I see it just throws a `sep` in there for the abs case, but it's not correct.

For windows, this need to turn a path like `In (In Init (Name "C")) (Name "Documents")` into `C:\Documents` (and `Init` into `\`).

My fault for not adding tests!

Reply:
✅